gnosis/cowswap
# [presign] Improve ux: consolidate order and tx

# Summary
The idea is of this PR is:
- Try to consolidate the presign transaction, and the order into one. So now, only the order is created.
- Then it will check the order status periodically, until it changes it's state from `presignaturePending` to `open`

This is how an order that has not been pre-sign is shown:
![image](https://user-images.githubusercontent.com/2352112/131895187-044e32e0-c8c1-4d6b-8062-7efdc8069e59.png)

After some seconds, when the transaction is mined, then it will show as an open order:
![image](https://user-images.githubusercontent.com/2352112/131895209-1e57665a-0eb5-42fe-ab71-000670bf7bfa.png)

After this, it follows the normal workflow. The order can be either fullfiled or 

## Not included

* [ ] Remove the cancelations
* [ ] Nicer styles, @michelbio u might want to create a PR and add another image if you want. For now, i just added react-feather
* [ ] Show a toast when the orders is signed. Still not working, but since the PR is big enough, i will address in another one. 
* [ ] The Toast for order creation, doesn't show the order information correctly. It just prints the order id. I will address this in another PR
![image](https://user-images.githubusercontent.com/2352112/131895907-96662d7a-495d-492a-92f0-22616ff829af.png)

# To Test

1. Create orders
2. Verify that transitions from the states described above
